.github/workflows/deploy-to-gcs.yml

```diff
@@ -0,0 +1,50 @@
+name: Deploy repository to GCS
+
+on:
+  workflow_call:
+    inputs:
+      gcp_workload_identity_provider:
+        required: true
+        type: string
+      gcp_service_account:
+        required: true
+        type: string
+
+permissions: {}
+
+jobs:
+  deploy-to-gcs:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: 'write' # For authenticating with the GitHub workflow identity
+
+    steps:
+      - uses: actions/download-artifact@eaceaf801fd36c7dee90939fad912460b18a1ffe # v4.1.2
+        with:
+          name: github-pages
+
+      - name: Prepare data for upload
+        run: |
+          # Extract the github-pages arcive into ./repository/
+          mkdir repository
+          tar --directory repository -xvf artifact.tar
+
+      - uses: google-github-actions/auth@a6e2e39c0a0331da29f7fd2c2a20a427e8d3ad1f # v2.1.1
+        with:
+          token_format: access_token
+          workload_identity_provider: ${{ inputs.gcp_workload_identity_provider }}
+          service_account: ${{ inputs.gcp_service_account }}
+
+      - uses: google-github-actions/setup-gcloud@98ddc00a17442e89a24bbf282954a3b65ce6d200 # v2.1.0
+        with:
+          project_id: projectsigstore-staging
+
+      - name: Upload repository to GCS
+        run: |
+          # Upload metadata, make sure we upload timestamp last
+          mv repository/timestamp.json .
+          gcloud storage cp --cache-control=no-store -r repository/* gs://tuf-root-staging/
+          gcloud storage cp --cache-control=no-store timestamp.json gs://tuf-root-staging/
+
```

Comment:
We can use `rsync` now - https://cloud.google.com/sdk/gcloud/reference/storage/rsync - to support cleaning up old files. Looks like this was recently added, since this used to only be in `gsutil`. It looks like it can also skip files, so we can do something like:

```
gcloud storage rsync repository/* gs://tuf-root-staging/ --recursive --cache-control=no-store --exclude="repository/timestamp.json"
gcloud storage cp --cache-control=no-store timestamp.json gs://tuf-root-staging/
```

Reply:
this looks great, I'll test this and modify the PR